bittensor/_subtensor/chain_data.py

```diff
@@ -16,28 +16,134 @@
+                ["validator_epoch_length", "Compact<u16>"],
+                ["max_allowed_validators", "Compact<u16>"],
+                ["min_allowed_weights", "Compact<u16>"],
+                ["max_weights_limit", "Compact<u16>"],
+                ["scaling_law_power", "Compact<u16>"],
+                ["synergy_scaling_law_power", "Compact<u16>"],
+                ["subnetwork_n", "Compact<u16>"],
+                ["max_allowed_uids", "Compact<u16>"],
+                ["blocks_since_last_step", "Compact<u64>"],
+                ["tempo", "Compact<u16>"],
+                ["network_modality", "Compact<u16>"],
+                ["network_connect", "Vec<[Compact<u16>; 2]>"],
+                ["emission_values", "Compact<u64>"]
+            ]
+        },
+        "DelegateInfo": {
+            "type": "struct",
+            "type_mapping": [
+                ["delegate_ss58", "AccountId"],
+                ["take", "Compact<u16>"],
+                ["nominators", "Vec<(AccountId, Compact<u64>)>"],
+                ["owner_ss58", "AccountId"]
+            ],
+        },
+        "NeuronInfo": {
+            "type": "struct",
+            "type_mapping": [
+                ["hotkey", "AccountId"],
+                ["coldkey", "AccountId"],
+                ["uid", "Compact<u16>"],
+                ["netuid", "Compact<u16>"],
+                ["active", "bool"],
+                ["axon_info", "AxonInfo"],
+                ["prometheus_info", "PrometheusInfo"],
+                ["stake", "Vec<(AccountId, Compact<u64>)>"],
+                ["rank", "Compact<u16>"],
+                ["emission", "Compact<u64>"],
+                ["incentive", "Compact<u16>"],
+                ["consensus", "Compact<u16>"],
+                ["trust", "Compact<u16>"],
+                ["validator_trust", "Compact<u16>"],
+                ["dividends", "Compact<u16>"],
+                ["last_update", "Compact<u64>"],
+                ["validator_permit", "bool"],
+                ["weights", "Vec<(Compact<u16>, Compact<u16>)>"],
+                ["bonds", "Vec<(Compact<u16>, Compact<u16>)>"],
+                ["pruning_score", "Compact<u16>"]
+            ],
+        },
+        "AxonInfo": {
+            "type": "struct",
+            "type_mapping": [
+                ["block", "u64"],
+                ["version", "u32"],
+                ["ip", "u128"],
+                ["port", "u16"],
+                ["ip_type", "u8"],
+                ["protocol", "u8"],
+                ["placeholder1", "u8"],
+                ["placeholder2", "u8"],
+            ],
+        },
+        "PrometheusInfo": {
+            "type": "struct",
+            "type_mapping": [
+                ["block", "u64"],
+                ["version", "u32"],
+                ["ip", "u128"],
+                ["port", "u16"],
+                ["ip_type", "u8"],
+            ],
+        },
+    }   
+}
+
+class ChainDataType(Enum):
+    NeuronInfo = 1
+    SubnetInfo = 2
+    DelegateInfo = 3
```

Comment:
Does this class need to be extended to the axon and prometheus info as well?

Reply:
those don't get sent by themselves on rpc so they don't need the enum for decoding